src/interfaces/IMetaMorpho.sol

```diff
@@ -30,57 +30,148 @@ interface IOwnable {
     function guardian() external view returns (address);
 
+    /// @notice The current fee.
     function fee() external view returns (uint96);
+
+    /// @notice The fee recipient.
     function feeRecipient() external view returns (address);
+
+    /// @notice The skim recipient.
     function skimRecipient() external view returns (address);
+
+    /// @notice The current timelock.
     function timelock() external view returns (uint256);
+
+    /// @dev Stores the order of markets on which liquidity is supplied upon deposit.
+    /// @dev Can contain any market. A market is skipped as soon as its supply cap is reached.
     function supplyQueue(uint256) external view returns (Id);
+
+    /// @notice Returns the length of the supply queue.
     function supplyQueueLength() external view returns (uint256);
+
+    /// @dev Stores the order of markets from which liquidity is withdrawn upon withdrawal.
+    /// @dev Always contain all non-zero cap markets as well as all markets on which the vault supplies liquidity,
+    /// without duplicate.
     function withdrawQueue(uint256) external view returns (Id);
+
+    /// @notice Returns the length of the withdraw queue.
     function withdrawQueueLength() external view returns (uint256);
 
+    /// @notice Stores the total assets managed by this vault when the fee was last accrued.
+    /// @dev May be a little off `totalAssets()` after each interaction, due to some roundings.
     function lastTotalAssets() external view returns (uint256);
 
+    /// @notice Submits a `newTimelock`.
+    /// @dev In case the new timelock is higher than the current one, the timelock is set immediately.
+    /// @dev Warning: Submitting a timelock will overwrite the current pending timelock.
     function submitTimelock(uint256 newTimelock) external;
+
+    /// @notice Accepts the pending timelock.
     function acceptTimelock() external;
+
+    /// @notice Revokes the pending timelock.
     function revokePendingTimelock() external;
 
-    function submitCap(MarketParams memory marketParams, uint256 supplyCap) external;
+    /// @notice Submits a `newSupplyCap` for the market defined by `marketParams`.
+    /// @dev In case the new cap is lower than the current one, the cap is set immediately.
+    /// @dev Warning: Submitting a cap will overwrite the current pending cap.
+    function submitCap(MarketParams memory marketParams, uint256 newSupplyCap) external;
+
+    /// @notice Accepts the pending cap of the market defined by `id`.
     function acceptCap(Id id) external;
+
+    /// @notice Revokes the pending cap of the market defined by `id`.
     function revokePendingCap(Id id) external;
 
+    /// @notice Submits a forced market removal from the vault, potentially losing all funds supplied to the market.
+    /// @dev Warning: Submitting a forced removal will overwrite the timestamp at which the market will be removable.
     function submitMarketRemoval(Id id) external;
```

Comment:
It could be worth specifying that this function does set the supply cap to `0` immediately, but does not remove such market from the withdraw queue. You need to later on call `updateWithdrawQueue` to remove it from the queue.